### PR TITLE
retrieve dynamic search for query request

### PR DESCRIPTION
**Pull request by stephherbers**
## Product Description

Core element of this ticket: [here](https://dimagi-dev.atlassian.net/browse/USH-3666)

formplayer ticket [here](https://github.com/dimagi/formplayer/pull/1476)
HQ Part one [here](https://github.com/dimagi/commcare-hq/pull/33622) and Part 2 [here](https://github.com/dimagi/commcare-hq/pull/33623)

will update ref shortly

## Technical Summary

see [this ticket](https://github.com/dimagi/commcare-hq/pull/33622)

### Safety story

No logic change on formplayer, just sends back the variable if a QueryResponse

### Automated test coverage

I will add test shortly 

### QA Plan

I am not requesting QA

### Special deploy instructions

- [X] This PR can be deployed after merge with no further considerations.

### Rollback instructions

- [ ] This PR can be reverted after deploy with no further considerations.
there is a formplayer PR as well linked above

### Review

- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change.

### Duplicate PR
Automatically duplicate this PR as defined in [contributing.md](https://github.com/dimagi/commcare-core/blob/master/.github/contributing.md).

### CODE REVIEW

File: src/cli/java/org/commcare/util/screen/QueryScreen.java [hunk 1]
```diff
@@ -62,6 +62,8 @@ public class QueryScreen extends Screen {
 
     private boolean defaultSearch;
 
+    private boolean dynamicSearch;
```
Comment (shubham1g5) [inline src/cli/java/org/commcare/util/screen/QueryScreen.java:65]: How does this get set for `QueryScreen` ?

Comment (stephherbers) [reply src/cli/java/org/commcare/util/screen/QueryScreen.java:65]: https://github.com/dimagi/commcare-core/pull/1358/commits/8bfbc6f54aa0b5eb32a1c7c493cafa7022cb0390

File: src/main/java/org/commcare/suite/model/RemoteQueryDatum.java [hunk 2]
```diff
@@ -102,7 +108,7 @@ public void readExternal(DataInputStream in, PrototypeFactory pf)
         description = (Text) ExtUtil.read(in, new ExtWrapNullable(Text.class), pf);
         useCaseTemplate = ExtUtil.readBool(in);
         defaultSearch = ExtUtil.readBool(in);
-        
+        dynamicSearch = ExtUtil.readBool(in);
```
Comment (shubham1g5) [inline src/main/java/org/commcare/suite/model/RemoteQueryDatum.java:111]: you need to add the corresponding `write` in `writeExternal` as well

Comment (stephherbers) [reply src/main/java/org/commcare/suite/model/RemoteQueryDatum.java:111]: [d35d47f](https://github.com/dimagi/commcare-core/pull/1358/commits/d35d47f54e54849a44d0a7341ee5b1568e97b901)

File: src/main/java/org/commcare/suite/model/RemoteQueryDatum.java [hunk 3]
```diff
@@ -81,6 +83,10 @@ public boolean doDefaultSearch() {
         return defaultSearch;
     }
 
+    public boolean getDynamicSearch() {
```
Comment (shubham1g5) [inline src/main/java/org/commcare/suite/model/RemoteQueryDatum.java:86]: nit: rename boolean getters in this PR to  `is{Var}` to signify it's a boolean

**Review by shubham1g5 — APPROVED** [review]

**Comment by shubham1g5** [comment]
duplicate this PR 955eb3ae506de6244c5fc0e7515a4843dfe65c6f e0e60cde51a29d7697a2472b3c1c08d0c0292304

**Pull request merged**